cf-custom-resources/lib/alb-rule-priority-generator.js

```diff
@@ -158,22 +158,41 @@ const getListenerRules = async function (listenerArn) {
  */
 exports.nextAvailableRulePriorityHandler = async function (event, context) {
   let responseData = {};
+  let nextRootRuleNumber, nextNonRootRuleNumber;
   const physicalResourceId =
-    event.PhysicalResourceId || `alb-rule-priority-${event.LogicalResourceId}`;
-  let isRootPath = event.ResourceProperties.RulePath === "/";
-
+      event.PhysicalResourceId || `alb-rule-priority-${event.LogicalResourceId}`;
   try {
     switch (event.RequestType) {
       case "Create":
       case "Update":
-        if (isRootPath) {
-          responseData.Priority = await calculateNextRootRulePriority(
-            event.ResourceProperties.ListenerArn
-          );
+        if (event.ResourceProperties.RulePath.length == 1){
```

Comment:
Why do we take `event.ResourceProperties.RulePath.length == 1` as a special case? Or will we combine them later?

Reply:
I was thinking to keep one port vs multiple port implementation separate. I thought of combining them into one and then we will have this following logic -

```

for (let i=0; i < event.ResourceProperties.RulePath.length; i++){
            if (event.ResourceProperties.RulePath[i] === "/") {
              if (nextRootRuleNumber == null){
                nextRootRuleNumber = await calculateNextRootRulePriority(
                    event.ResourceProperties.ListenerArn
                );
              }
              responseData["Priority"+i]  = nextRootRuleNumber--;
            } else {
              if (nextNonRootRuleNumber == null) {
                nextNonRootRuleNumber = await calculateNextRulePriority(
                    event.ResourceProperties.ListenerArn
                );
              }
              responseData["Priority"+i]  = nextNonRootRuleNumber++;
            }
          }
```

In this case we return `Prioirty0`, `Prioirty1`, `Prioirty2` and so on. But currently we only return `Priority` without any postfix.  But I think as per our offline discussion with Wx also whenever customer will upgrade to new version cfn template will also expect `Priority{{ $i }}` instead of just `Priority` and that shouldn't replace listener rule all together.

Reply:
But as you said for this PR I think we need to keep it as I have implemented because we do not have `Priority{{ $i }}` logic implemented yet. So removing `event.ResourceProperties.RulePath.length == 1` will result into errors.

Reply:
Can we still return Prioirty, Prioirty1, Prioirty2 if we add an "if statement" before `responseData["Priority"+i]  = nextNonRootRuleNumber++;`?

Reply:
I can do that but I think I need to do the same if for `responseData["Priority"+i]  = nextRootRuleNumber--;` as well.